spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java

```diff
@@ -365,6 +369,16 @@ public static SparkBigQueryConfig from(
               + config.cacheExpirationTimeInMinutes);
     }
 
+    String traceIdParam =
+        getAnyOption(globalOptions, options, "traceId").or(SparkBigQueryUtil.getJobId(sqlConf));
+    if (traceIdParam.length() > MAX_TRACE_ID_LENGTH) {
```

Comment:
I think we do bytes, so this would this should probably encode in UTF-8 before checking if you want to be strict.

Reply:
The [client API](https://github.com/googleapis/java-bigquerystorage/blob/v2.11.0/proto-google-cloud-bigquerystorage-v1/src/main/java/com/google/cloud/bigquery/storage/v1/ReadSession.java#L5878) requests String

Reply:
right, but the String will be converted to UTF-8 first, so the string length here will not be accurate for non-ascii characters.  I think maybe either remove this check or do the check?